tests/test_managed_agents.py

```
@@ -0,0 +1,881 @@
+        to_money_string(Decimal("Infinity"))
+    # An exponent still within the wallet's range formats normally.
+    assert to_money_string(Decimal("1E+5")) == "100000"
+
+
+def test_unicode_digits_are_refused():
+    """Python's `\\d` matches every Unicode decimal digit, so "١٢" passed the
+    pattern and went to an API that reads ASCII only."""
+    with pytest.raises(ValueError, match="not a valid amount"):
+        to_money_string("١٢")
+    with pytest.raises(ValueError, match="not a valid amount"):
+        to_money_string("٥.٠٠")
```

Ruff (0.16.1)

[warning] String contains ambiguous `٥` (ARABIC-INDIC DIGIT FIVE). Did you mean `o` (LATIN SMALL LETTER O)?

(RUF001)